pallets/marketplace/src/lib.rs

```diff
@@ -409,6 +410,13 @@ pub mod pallet {
                 // either all or no candidate gets assigned after checking if all slots can be filled
                 let mut candidates = Vec::new();
                 for ad_with_reward in ads {
+                    // CHECK attestation
+                    if registration.allow_only_verified_sources
```

Comment:
We probably want to use `ensure_source_allowed` here, right?

Reply:
that's not needed, later on the whitelist is also checked, which is what ensure_source_allowed also does in addition to the attestation check